### ISSUE TITLE
clarifying questions on tuning dynamical HPs like learning rate 

### ISSUE THREAD

**Issue by yzhao062** [pomonam/Self-Tuning-Networks#3]
Hi Juhan,

Thank you so much for the codebase. Very useful in reproducing the paper(s) and exploring new directions.

I understand the case that dynamical HPs like learning rate cannot be tuned since they are not part of the validation loss. 
![image](https://user-images.githubusercontent.com/15079146/188776945-47dadde2-209a-4e09-9875-011df61ab4e8.png)

Just wonder assume if I could use other ways to replace the validation loss to find the next promising HPs and scales of any HPs based on the current function, can I learn include the learning rate as part of the HPs to be tuned? 
![image](https://user-images.githubusercontent.com/15079146/188776921-e5e87deb-2915-4fb5-8828-dcc6757e1f2a.png)
The update of the response function appears not to depend on the \lambda ...

I am a bit confused here and wish for your input!
Thank you!

Yue

**Comment by yzhao062** [comment]
or if I put the question in another way...if we include some hyperparameters like learning rate in the h_net but does not use it in the loss (since learning rate is not part of the training and validation loss and we do not know how to use it), what will happen to the response function? I guess it will ignore the effect of the learning rate. 

Sorry for this stupid question... thank you so much!

**Comment by pomonam** [comment]
Hi Yue,

Thank you for the issue! Yes, you are correct that optimization parameters (such as learning rate) cannot be tuned as it does not affect the validation loss. If you include the learning rate in the h_net, I also suspect that it will simply ignore the effect of the learning rate (the corresponding learnable parameters can be just 0). 

Hope this answers your question.

Juhan

**Comment by yzhao062** [comment]
Hi Juhan,

Thanks for the confirmation. Totally agreed with the point that including LR in the h_net will not play any role in the learning of the response function. Looking forward to your new work!

Cheers!
